src/components/productPrice/productPrice.jsx

```diff
@@ -13,40 +13,63 @@ const ProductPrice = ({
     discountPriceInCents,
     originalPriceInCents,
     minimalPrice,
-}) => (
-    <PriceText className={className}>
-        {minimalPrice && (
-            <span className="grover-product-price__starting-price-text">
-                <FormattedMessage translationKey="STARTING_PRICE" />
-            </span>
-        )}
-        {discountPriceInCents ? (
-            <DiscountPrice
-                className={cn(
-                    'grover-product-price__price-container',
-                    classNames.discountPriceContainer
-                )}
-                classNames={{
-                    discountPrice: classNames.discountPrice,
-                    originalPrice: classNames.originalPrice,
-                }}
-                discountPriceInCents={discountPriceInCents}
-                originalPriceInCents={originalPriceInCents}
-            />
-        ) : (
+    showOnlyDiscountedPrice,
+}) => {
+    const showPrice = () => {
+        if (discountPriceInCents) {
+            if (showOnlyDiscountedPrice) {
+                return (
+                    <Price
+                        className={cn(
+                            'grover-product-price__price-container',
+                            classNames.price
+                        )}
+                        priceInCents={discountPriceInCents}
+                    />
+                );
+            }
+
+            return (
+                <DiscountPrice
+                    className={cn(
+                        'grover-product-price__price-container',
+                        classNames.discountPriceContainer
+                    )}
+                    classNames={{
+                        discountPrice: classNames.discountPrice,
+                        originalPrice: classNames.originalPrice,
+                    }}
+                    discountPriceInCents={discountPriceInCents}
+                    originalPriceInCents={originalPriceInCents}
+                />
+            );
+        }
```

Comment:
Small one here, I think we can save one indent with an early return here by flipping the condition and returning the simpler one without a discount first, as well as checking for the new boolean, something like this 👍🏼 

```
if (!discountPriceInCents || showOnlyDiscountedPrice) {
  // return the one at the bottom
}

// the rest with discounted prices
```

WDYT? 😄

Reply:
@Nelspike the price is different... I can have the price in a variable tho.